opal/mca/common/ucx/common_ucx.c

```diff
@@ -139,17 +139,27 @@ void opal_common_ucx_mca_proc_added(void)
 {
 #if HAVE_DECL_UCM_TEST_EVENTS
     static int warned = 0;
-    static char *mem_hooks_suggestion = "Pls try adding --mca opal_common_ucx_opal_mem_hooks 1 "
-                                        "to mpirun/oshrun command line to resolve this issue.";
+    static const char *mem_hooks_suggestion =
+        "Pls try adding --mca opal_common_ucx_opal_mem_hooks 1 "
+        "to mpirun/oshrun command line to resolve this issue.";
     ucs_status_t status;
 
     if (!warned) {
-        status = ucm_test_events(UCM_EVENT_VM_UNMAPPED);
+#if HAVE_DECL_UCM_TEST_EXTERNAL_EVENTS
+        if (opal_common_ucx.opal_mem_hooks) {
+            status     = ucm_test_external_events(UCM_EVENT_VM_UNMAPPED);
+        } else
+#endif
+        {
+            status     = ucm_test_events(UCM_EVENT_VM_UNMAPPED);
+        }
+
         if (status != UCS_OK) {
             MCA_COMMON_UCX_WARN("UCX is unable to handle VM_UNMAP event. "
                                 "This may cause performance degradation or data "
                                 "corruption. %s",
-                                opal_common_ucx.opal_mem_hooks ? "" : mem_hooks_suggestion);
+                                opal_common_ucx.opal_mem_hooks ?
+                                "" : mem_hooks_suggestion);
```

Comment:
maybe set `mem_hooks_suggestion = "Please check OPAL memory events infrastructure"`

Reply:
ok, done: added one more suggestion

Reply:
don't see it..

Reply:
it github freezes :)
try to refresh browser tab